### DISCUSSION TITLE
Add Windows support

### DISCUSSION THREAD

**Discussion by lucassz** [borgbase/vorta#824] [Packaging]
Documentation mentions that Vorta could support Windows, but that Borg doesn't. Apparently [now it does](https://github.com/borgbackup/borg/issues/936) -- I'm not sure if the documentation just needs to be changed, or if any small changes need to be made in order for Vorta to support Windows.

**Comment by Hofer-Julian** [comment]
To me this reads more like there are people currently working on it. Not at all like borg already supports Windows.

**Comment by ThomasWaldmann** [comment]
Yeah, as @Hofer-Julian says.

borg does not support running under native windows yet, but it works within cygwin or WSL.

**Comment by FabianKnapp** [comment]
> Yeah, as @Hofer-Julian says.
> 
> borg does not support running under native windows yet, but it works within cygwin or WSL.

So is it possible to backup files on windows NTFS partition within WSL?

**Comment by m3nu** [comment]
Should be possible, as Windows makes all drives accessible under `/mnt/c`, etc.

It's on my todo-list to actually test it but from what I saw on Borg's Github it should work.

Found some more details here: https://superuser.com/questions/1114341/windows-10-ubuntu-bash-shell-how-do-i-mount-other-windows-drives

**Comment by erotavlasme** [comment]
Borg 1.0.11 already works well under windows 10 64 bit. It is available via [chocolatey package](https://chocolatey.org/packages/borgbackup).

**Comment by Ramblurr** [comment]
I've done some basic tests of Borg on windows 10 via chocolatey, and it works fine. Next step is definitely getting a Vorta windows build, then I can retire the other kludgy backup systems my family uses.

**Comment by ThomasWaldmann** [comment]
Seems like the chocolatey borg is still at 1.0.10, which is rather old.

The borg collective is still working on native windows support (not released yet).

Current borg releases (preferably 1.1.latest) can be used with cygwin or wsl on windows machines.

**Comment by m3nu** [comment]
Some changes related to this: https://www.theverge.com/2020/5/19/21263377/microsoft-windows-10-linux-gui-apps-gpu-acceleration-wsl-features

**Comment by jucor** [comment]
@Ramblurr  , do you know who's the maintainer of the Borg package on chocolatey, please? Could you tag them here, maybe?

**Comment by JBlond** [comment]
The link is https://package.chocolatey.org/profiles/ARCHer

**Comment by jucor** [comment]
Thanks @JBlond . Sorry, it was right there on the page, I should have seen that. I've contacted the maintainer, pointing him to this thread, see if he wants to chime in and/or update his package.

**Comment by ojaksch** [comment]
I'm the maintainer of borgbackup for chocolatey but not coder of it, which is [Billy Charlton](https://github.com/billyc/borg-releases/).
I did the choco package because of "better old than nothing" - we should whether wait for an official version of borgbackup for windows or ask Billy/someone to update it. Would be glad to update choco's package for you Windows users.

**Comment by m3nu** [comment]
Cool. With a working Borg.exe, we can probably package Vorta (which is written in Qt) for Windows eventually.

**Comment by m3nu** [comment]
You're still updating this repo @billyc?

**Comment by jucor** [comment]
@DanielSzoska is this something you still develop? Daniel forked @billyc's unofficial windows releases and updated them, I believe using cygwin as @ThomasWaldmann mentioned.

**Comment by JBlond** [comment]
@jucor The base for that build is cygwin
The most recent working version that I found is https://github.com/engelant/borg-cygwin
@engelant are you still working on this?

**Comment by ojaksch** [comment]
As long as there's no (current) native Windows port of borg, what about using the original and current version of borg with Microsoft's WSL1 (or 2) and a Linux distrubution of your choice? It doesn't matter if one installs a big cygwin or a big "real" Linux setup imho to gain access of borg power.
I'm a Linux only user and am out of Microsoft's scope since years, so that's my thoughts only.

**Comment by m3nu** [comment]
A fellow user – Christer – has shared the following instructions on setting up WSL2 by email and gave permission to use them for instructions, if we ever need to. I haven't found a good place for them yet, but if someone can validate, we can put it in the Vorta docs somewhere or even the Borg docs.

> A few things worth noting, if you're gonna be testing and/or trying to turn this into anything others can use:
> 
> * "wsl --set-default-version 2" will only work in build 19041 onward. If you run an earlier build than that, either skip that command (and keep running on WSL 1 - the command will just fail with an error), or upgrade Win10 first (build 19041 should now be available to anyone, the latest as of writing, since I believe May).
> 
> * The order of the above wsl command is assumed, since I haven't yet had a chance to test this with WSL 2 out of the box, only upgraded WSL 1. If you're coming from WSL 1 as I did, you can instead retroactively do this to upgrade an existing Debian VM:
> - wsl --set-version Debian 2
> - (optional for any future installs) wsl --set-default-version 2
> 
> * Enabling the features can supposedly also be done sans PowerShell (though I haven't tried myself):
> 
> dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart
> 
> (and similar for /featurename:Microsoft-Windows-Subsystem-Linux I imagine)
> If you also found some alternative way to specify, download and install an image, then you could drop the PowerShell dependency. The Invoke-WebRequest call was just the closest thing I could find to Wget/cURL as I was writing my notes.
> 
> * I've diligently followed official instructions to reboot after each of the first two steps, though it's quite possible that all steps can be completed with a single reboot at the end (see the /norestart flag in the non-PS example), in which case this could be scripted easily. Seems worth testing.
> 
>

**Comment by Ramblurr** [comment]
FWIW I'm all for a WSL solution if that's what we need at first to get Vorta on windows.

But a non-WSL solution is much more valuable. It will open Vorta and Borg to a huge number of workstations whose users and admins can't use WSL

**Comment by Eeems** [comment]
So the website is a little misleading. The install page has a large buttons saying windows, which you have to click to find out that it's not supported yet. The page says that support is coming soon, but I see that this issue is in a planning state, and has been in that state since June 2019.

What is the current status of this, and should the website be made clearer on what the state (or lack thereof) of windows support is for Vorta?

**Comment by m3nu** [comment]
Vorta would run OK on Windows, but we would need a Windows binary to use. Those exist, but aren't updated regularly.

**Comment by m3nu** [comment]
There are also people running it on WSL2. Someone even sent me instructions for it. Maybe we put those on the Windows page for now?

**Comment by Eeems** [comment]
I managed to get it running in WSL as well (https://eeems.website/running-vorta-on-windows/). Likely having some sort of instructions linked would be good. That said, still having an actual answer on when/if native windows support is in the works would be ideal.

**Comment by m3nu** [comment]
Updated the docs and also included your blog post and some more details added by Christer a few months ago. He used Debian instead of Ubuntu.

If someone can update those instructions, let's do it in the [docs repo](https://github.com/borgbase/vorta.borgbase.com).
